ERCS/erc-7690.md

```diff
@@ -0,0 +1,129 @@
+---
+eip: 7690
+title: Union Lock
+description: Union Lock based on TSTORE/TLOAD can avoid flash loan attacks.
+author: Elon Lee (@1999321)
+discussions-to: https://ethereum-magicians.org/t/union-lock-based-on-tstore-tload-can-avoid-flash-loan-attacks/19676
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-04-16
+requires: 8000
+---
+
+## Abstract
+
+The feature of Union Lock can query the number of calls of other contracts and the number of calls of other contract functions in the same Ethereum transaction, thereby realizing the cross-contract locking function.
+
+Union Lock is based on  TSTORE/TLOAD, it can make DApps avoid flash loan attacks.
+
+## Motivation
+
+Since the losses caused by flash loan attacks are as high as tens of millions of dollars, it would be meaningful if a design could be provided to enable DApps to avoid flash loan attacks.
+
+## Specification
+
+```solidity
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.25;
+
+interface IERC7690 {
+
+    function getFunctionCallTimes(bytes4 selector) external view returns (uint256);
+
+    function getContractCallTimes() external view returns (uint256);
+}
+```
+
+`getFunctionCallTimes`: Allows cross-contract acquisition of the number of times a function is called in a transaction.
+
+`getContractCallTimes`: Allows cross-contract acquisition of the number of times a contract is called in a transaction.
+
+## Rationale
```

Comment:
All of your rationale (except the last paragraph) seems to be justifying the proposal as a whole. That kind of content belongs in the Motivation section.

The Rationale section exists to explain choices within the proposal.

My favorite analogy is:

> Motivation: We need to build a shed because...
> Rationale: We painted the shed red because...

Reply:
Move this part  (except the last paragraph) to rationale. change it to "Based on [[EIP1153](https://github.com/ethereum/ERCs/pull/382/eip-1153.md)](./eip-1153.md),  Union Lock can count the number of contract calls and function calls with TSTORE, and if it is accessed across contracts in the same transaction, it will be automatically reset to zero in the next transaction. In the same transaction, you can know the number of function accesses of any contract that uses Union Lock and the number of accesses of the contract, so as to formulate corresponding strategies.Based on [[EIP1153](https://github.com/ethereum/ERCs/pull/382/eip-1153.md)](./eip-1153.md),  Union Lock can count the number of contract calls and function calls with TSTORE, and if it is accessed across contracts in the same transaction, it will be automatically reset to zero in the next transaction. In the same transaction, you can know the number of function accesses of any contract that uses Union Lock and the number of accesses of the contract, so as to formulate corresponding strategies."